lisa/sut_orchestrator/azure/features.py

```diff
@@ -1749,40 +1794,69 @@ def get_luns(self) -> Dict[str, int]:
                 device_luns.update({cmd_result.stdout: device_lun})
         return device_luns
 
-    def get_raw_data_disks(self) -> List[str]:
-        # Handle BSD case
-        if isinstance(self._node.os, BSD):
-            return self._get_raw_data_disks_bsd()
-
-        # disk_controller_type == NVME
+    def get_nvme_os_disk_controller(self) -> str:
+        # Getting OS disk nvme controller.
+        # Sample os_boot_partition:
+        # name: /dev/nvme0n1p15, disk: nvme, mount_point: /boot/efi, type: vfat
+        # In the above example, '/dev/nvme0' is the controller.
+        os_disk_controller = ""
         node_disk = self._node.features[Disk]
         if node_disk.get_os_disk_controller_type() == schema.DiskControllerType.NVME:
-            # Getting OS disk nvme namespace and disk controller used by OS disk.
-            # Sample os_boot_partition:
-            # name: /dev/nvme0n1p15, disk: nvme, mount_point: /boot/efi, type: vfat
             os_boot_partition = node_disk.get_os_boot_partition()
             if os_boot_partition:
-                os_disk_namespace = get_matched_str(
-                    os_boot_partition.name,
-                    self.NVME_NAMESPACE_PATTERN,
-                )
                 os_disk_controller = get_matched_str(
                     os_boot_partition.name,
                     self.NVME_CONTROLLER_PATTERN,
                 )
+        return os_disk_controller
+
+    def get_nvme_os_disk_namespace(self) -> str:
+        # Getting OS disk nvme namespace.
+        # Sample os_boot_partition:
+        # name: /dev/nvme0n1p15, disk: nvme, mount_point: /boot/efi, type: vfat
+        # In the above example, '/dev/nvme0n1' is the namespace.
+        os_disk_namespace = ""
+        node_disk = self._node.features[Disk]
+        if node_disk.get_os_disk_controller_type() == schema.DiskControllerType.NVME:
+            os_boot_partition = node_disk.get_os_boot_partition()
+            if os_boot_partition:
+                os_disk_namespace = get_matched_str(
+                    os_boot_partition.name,
+                    self.NVME_NAMESPACE_PATTERN,
+                )
+        return os_disk_namespace
+
+    def _is_remote_data_disk(self, disk: str) -> bool:
+        # If disk_controller_type == NVME
+        node_disk = self._node.features[Disk]
```

Comment:
it's the `self`, it doesn't need to get its _node, and get itself again. Just call `self.get_os_disk_controller_type()`

Reply:
Fixed